### PR TITLE
Add more clarity to the rate-limiting docs

### PR DESCRIPTION
**Pull request by varsanojidan**
How we configure rate-limiting needs some work 😛. But at least this should clarify things.

## Test plan

docs change

### CODE REVIEW

File: doc/admin/repo/update_frequency.md [hunk 1]
```diff
@@ -8,20 +8,24 @@ Repositories will never be updated more frequently than 45 seconds, and no less
 
 After Sourcegraph has updated a repository's Git data, the global search index will automatically update a short while after (usually a few minutes).
 
-## Limiting repository updates
+## Rate Limiting
+If you wish to control how frequently repositories are discovered or how frequently Sourcegraph polls your code host for updates, the following options are available:
 
-If you wish to control how frequently repositories are discovered or how frequently Sourcegraph polls your code host for updates, tuning parameters are available in the site configuration:
+### Code host API requests
```
Comment (sashaostrikov) [inline doc/admin/repo/update_frequency.md:14]: maybe?
```suggestion
### Limiting a number or code host API requests
```

File: doc/admin/repo/update_frequency.md [hunk 2]
```diff
@@ -8,20 +8,24 @@ Repositories will never be updated more frequently than 45 seconds, and no less
...
+All other code hosts that don't support explicitly setting the rate limit in the configuration, default to a rate limit of: [defaultRateLimit](../config/site_config.md#defaultRateLimit) requests per hour.
 
-You may also choose to disable automatic Git updates entirely and instead [configure repository webhooks](webhooks.md).
+- Site config: [repoListUpdateInterval](../config/site_config.md#repoListUpdateInterval) controls how frequently we check the code host _for new repositories_ in minutes.
 
-## Code host API rate limiting
+**NOTE** Internal rate limiting is currently only enforced for syncing changesets in [batch changes](../../batch_changes/index.md), repository permissions and repository metadata from code hosts.
 
-Sourcegraph uses a configurable internal rate limiter for API requests made from Sourcegraph to [GitHub](../external_service/github.md#internal-rate-limits), [GitLab](../external_service/gitlab.md#internal-rate-limits), [Bitucket Server](../external_service/bitbucket_server.md#internal-rate-limits) and [Bitbucket Cloud](../external_service/bitbucket_cloud.md#internal-rate-limits).
+### Code host SSH requests
```
Comment (sashaostrikov) [inline doc/admin/repo/update_frequency.md:23]: maybe [2]?
```suggestion
### Limiting a number of code host SSH requests
```

File: doc/admin/repo/update_frequency.md [hunk 3]
```diff
@@ -8,20 +8,24 @@ Repositories will never be updated more frequently than 45 seconds, and no less
...
-## Limiting repository updates
+## Rate Limiting
+If you wish to control how frequently repositories are discovered or how frequently Sourcegraph polls your code host for updates, the following options are available:
 
-If you wish to control how frequently repositories are discovered or how frequently Sourcegraph polls your code host for updates, tuning parameters are available in the site configuration:
+### Code host API requests
 
-- [repoListUpdateInterval](../config/site_config.md#repoListUpdateInterval) controls how frequently we check the code host _for new repositories_ in minutes.
-- [gitMaxConcurrentClones](../config/site_config.md#gitMaxConcurrentClones) controls the maximum number of _concurrent_ cloning / pulling operations per gitserver that Sourcegraph will perform.
+- Code host config: Sourcegraph uses a configurable internal rate limiter for API requests made from Sourcegraph to [GitHub](../external_service/github.md#internal-rate-limits), [GitLab](../external_service/gitlab.md#internal-rate-limits), [Bitucket Server](../external_service/bitbucket_server.md#internal-rate-limits) and [Bitbucket Cloud](../external_service/bitbucket_cloud.md#internal-rate-limits).
```
Comment (sashaostrikov) [inline doc/admin/repo/update_frequency.md:16]: I guess we prefer full version in docs:
```suggestion
- Code host configuration: Sourcegraph uses a configurable internal rate limiter for API requests made from Sourcegraph to [GitHub](../external_service/github.md#internal-rate-limits), [GitLab](../external_service/gitlab.md#internal-rate-limits), [Bitucket Server](../external_service/bitbucket_server.md#internal-rate-limits) and [Bitbucket Cloud](../external_service/bitbucket_cloud.md#internal-rate-limits). 
```

File: doc/admin/repo/update_frequency.md [hunk 4]
```diff
@@ -8,20 +8,24 @@ Repositories will never be updated more frequently than 45 seconds, and no less
...
-- [gitMaxConcurrentClones](../config/site_config.md#gitMaxConcurrentClones) controls the maximum number of _concurrent_ cloning / pulling operations per gitserver that Sourcegraph will perform.
+- Code host config: Sourcegraph uses a configurable internal rate limiter for API requests made from Sourcegraph to [GitHub](../external_service/github.md#internal-rate-limits), [GitLab](../external_service/gitlab.md#internal-rate-limits), [Bitucket Server](../external_service/bitbucket_server.md#internal-rate-limits) and [Bitbucket Cloud](../external_service/bitbucket_cloud.md#internal-rate-limits). 
+All other code hosts that don't support explicitly setting the rate limit in the configuration, default to a rate limit of: [defaultRateLimit](../config/site_config.md#defaultRateLimit) requests per hour.
 
-You may also choose to disable automatic Git updates entirely and instead [configure repository webhooks](webhooks.md).
+- Site config: [repoListUpdateInterval](../config/site_config.md#repoListUpdateInterval) controls how frequently we check the code host _for new repositories_ in minutes.
```
Comment (sashaostrikov) [inline doc/admin/repo/update_frequency.md:19]: here as well
```suggestion
- Site configuration: [repoListUpdateInterval](../config/site_config.md#repoListUpdateInterval) controls how frequently we check the code host _for new repositories_ in minutes.
```

File: doc/admin/repo/update_frequency.md [hunk 5]
```diff
@@ -8,20 +8,24 @@ Repositories will never be updated more frequently than 45 seconds, and no less
...
 
-## Code host API rate limiting
+**NOTE** Internal rate limiting is currently only enforced for syncing changesets in [batch changes](../../batch_changes/index.md), repository permissions and repository metadata from code hosts.
 
-Sourcegraph uses a configurable internal rate limiter for API requests made from Sourcegraph to [GitHub](../external_service/github.md#internal-rate-limits), [GitLab](../external_service/gitlab.md#internal-rate-limits), [Bitucket Server](../external_service/bitbucket_server.md#internal-rate-limits) and [Bitbucket Cloud](../external_service/bitbucket_cloud.md#internal-rate-limits).
+### Code host SSH requests
 
-**NOTE** Internal rate limiting is currently only enforced for syncing changesets in [batch changes](../../batch_changes/index.md), repository permissions and repository metadata from code hosts.
+- [gitMaxCodehostRequestsPerSecond](../config/site_config.md#gitMaxCodehostRequestsPerSecond) controls how many code host git operations can be run per second, per gitserver.
```
Comment (sashaostrikov) [inline doc/admin/repo/update_frequency.md:25]: This sounded better to me, but may be wrong
```suggestion
- [gitMaxCodehostRequestsPerSecond](../config/site_config.md#gitMaxCodehostRequestsPerSecond) controls how many git operations can be run against a code host per second, per gitserver.
```

File: doc/admin/repo/update_frequency.md [hunk 6]
```diff
@@ -8,20 +8,24 @@ Repositories will never be updated more frequently than 45 seconds, and no less
...
+- [gitMaxConcurrentClones](../config/site_config.md#gitMaxConcurrentClones) controls the maximum number of _concurrent_ cloning / pulling operations per gitserver that Sourcegraph will perform.
```
Comment (sashaostrikov) [inline doc/admin/repo/update_frequency.md:26]: Officer Nit Pickleson here 👮🏻‍♂️: please remove your spaces for a _single-word-slashing_. Thanks, sir!
```suggestion
- [gitMaxConcurrentClones](../config/site_config.md#gitMaxConcurrentClones) controls the maximum number of _concurrent_ cloning/pulling operations per gitserver that Sourcegraph will perform.
```

**Review by sashaostrikov — COMMENTED** [review]
LGTM, left some nits (of course)

File: doc/admin/repo/update_frequency.md (hunk 3 above)
Comment (mrnugget) [inline doc/admin/repo/update_frequency.md:16]: Can you turn the code host links into a sub-list here? Easier to spot at a glance

File: doc/admin/repo/update_frequency.md [hunk 7]
```diff
@@ -8,20 +8,24 @@ Repositories will never be updated more frequently than 45 seconds, and no less
...
+- Code host config: Sourcegraph uses a configurable internal rate limiter for API requests made from Sourcegraph to [GitHub](../external_service/github.md#internal-rate-limits), [GitLab](../external_service/gitlab.md#internal-rate-limits), [Bitucket Server](../external_service/bitbucket_server.md#internal-rate-limits) and [Bitbucket Cloud](../external_service/bitbucket_cloud.md#internal-rate-limits). 
+All other code hosts that don't support explicitly setting the rate limit in the configuration, default to a rate limit of: [defaultRateLimit](../config/site_config.md#defaultRateLimit) requests per hour.
 
-You may also choose to disable automatic Git updates entirely and instead [configure repository webhooks](webhooks.md).
+- Site config: [repoListUpdateInterval](../config/site_config.md#repoListUpdateInterval) controls how frequently we check the code host _for new repositories_ in minutes.
 
-## Code host API rate limiting
+**NOTE** Internal rate limiting is currently only enforced for syncing changesets in [batch changes](../../batch_changes/index.md), repository permissions and repository metadata from code hosts.
```
Comment (mrnugget) [inline doc/admin/repo/update_frequency.md:21]: Maybe we can use something that's easier to grasp here?

```suggestion
**NOTE** Internal rate limiting is currently only enforced for HTTP requests to code hosts. That means it's used when, for example, syncing changesets in [batch changes](../../batch_changes/index.md), repository permissions and repository metadata from code hosts.
```

Also, I think you need: `> NOTE:` for it to render correctly. Check other docs.

File: doc/admin/repo/update_frequency.md (hunk 2 above)
Comment (mrnugget) [inline doc/admin/repo/update_frequency.md:23]: Is it always SSH? Can't users also configure `https`?

**Review by mrnugget — APPROVED** [review]
Agree with Sasha's comments. But other than that: nice! thanks for improving the docs!

File: doc/admin/repo/update_frequency.md (hunk 2 above)
Comment (varsanojidan) [reply doc/admin/repo/update_frequency.md:23]: I meant to say Git requests, but my brain short-circuited.

**Pull request merged**